lib/thesis/notifications.ex

```diff
@@ -0,0 +1,163 @@
+defmodule Thesis.Notifications do
+  @moduledoc """
+  Builds a map of Notifications that is passed to the client side. The key should be a
+  representation of the category/type of notification (this is used to determine where and how it's
+  displayed). The value is a list of notification strings.
+
+  Example of final map:
+  %{
+    "page-settings": [],
+    "add-page": [],
+    "import-export-restore": [],
+    "other": []
+  }
+  """
+
+  alias Thesis.Backup
+  import Thesis.Config
+  import Thesis.Utilities
+  import Ecto.Query, only: [limit: 2]
+
+  @allowed_notification_keys [
+    "page-settings",
+    "add-page",
+    "import-export-restore",
+    "other",
+    :page_settings,
+    :add_page,
+    :import_export_restore,
+    :other
+  ]
+
+  def all(page) do
+    %{
+      "page-settings" => [],
+      "add-page" => [],
+      "import-export-restore" => [],
+      "other" => []
+    }
+    |> notifications_from_host_app(notifications())
+    |> notifications_regarding_env(Mix.env)
+    |> notifications_regarding_page(page)
+  end
+
+  def notifications_regarding_page(acc, nil), do: acc
+  def notifications_regarding_page(acc, %_{}) do
+    acc
+  end
+
+  def notifications_regarding_env(acc, :test), do: acc
+  def notifications_regarding_env(acc, _) do
+    acc
+    |> notification_to_run_migration_for_version_0_2_1()
+  end
```

Comment:
Notifs that can be added and removed as Thesis development continues. The on that is there right now is concerning the LZString compression removal. Basically, if Thesis is upgraded and migrations are executed, page revisions will not work.